torch/csrc/jit/runtime/profiling_record.cpp

```diff
@@ -214,6 +233,31 @@ void ProfilingRecord::instrumentBlock(Block* block) {
           (needsProfiledInputs(n) || needsProfiledOutput(i->node()))) {
         insertShapeProfile(n, offset);
       }
+
+      if (i->type()->cast<OptionalType>() && hasGradSumToSizeUses(i)) {
+        auto opt_pn = createProfileOptionalNode(nullptr, {i});
+        std::function<void(Stack&)> optional_profiler = [this,
+                                                         opt_pn](Stack& stack) {
+          std::lock_guard<std::mutex> lock(this->mutex_);
+          // frame_id is unused
+          int64_t frame_id = 0;
+          pop(stack, frame_id);
+          IValue value;
+          pop(stack, value);
+          if (value.isNone()) {
+            opt_pn->i_(attr::num_none, opt_pn->i(attr::num_none) + 1);
+          } else {
+            opt_pn->i_(attr::num_present, opt_pn->i(attr::num_present) + 1);
+          }
+          push(stack, value);
+        };
+        opt_pn->setCallback(optional_profiler);
+        auto pno = opt_pn->addOutput();
+        pno->setType(i->type());
+        WithInsertPoint guard(i->node());
```

Comment:
nitpick do we really need a `WithInsertPoint guard` here since we use `insertAfter`?